assets/component-visual-display.css

```diff
@@ -0,0 +1,100 @@
+.visual-display {
+  --visual-display__size: min(2.4rem, 100%);
+  position: relative;
+  width: var(--visual-display__size);
+  border: 0.1rem solid rgba(var(--color-foreground), 0.2);
+  aspect-ratio: 1/1;
+}
+
+.visual-display.empty {
+  border-style: dashed;
+}
+
+.visual-display--style-swatch {
+  --visual-display__size: min(2.4rem, 100%);
+
+  border-radius: 100%;
+  overflow: hidden;
+}
+
+.visual-display--style-thumbnail {
+  --visual-display__size: min(9rem, 100%);
+
+  border-radius: var(--buttons-radius);
+  overflow: hidden;
+}
+
+.visual-display-parent--thumbnail {
+  outline: 0.05rem solid rgba(var(--color-foreground), 0.75);
+}
+
+/* Hover, active, and focus states */
+:where(
+    .visual-display-parent:hover .visual-display--style-swatch,
+    .visual-display-parent.active .visual-display--style-swatch,
+    .visual-display-parent:has(:focus-visible) .visual-display--style-swatch,
+    .visual-display-parent--thumbnail:hover,
+    .visual-display-parent--thumbnail.active,
+    .visual-display-parent--thumbnail:has(:focus-visible)
+  ) {
+  outline: 0 solid rgb(var(--color-foreground));
+  outline-offset: 0.2rem;
+}
+
+/* Active state */
+.visual-display-parent.active .visual-display--style-swatch,
+.visual-display-parent--thumbnail.active {
+  outline-width: 0.1rem;
+}
+
+/* Hover state */
+.visual-display-parent:hover .visual-display--style-swatch,
+.visual-display-parent--thumbnail:hover {
+  outline-width: 0.15rem;
+}
+
+/* Focus state */
+.visual-display-parent:has(:focus-visible) .visual-display--style-swatch,
+.visual-display-parent--thumbnail:has(:focus-visible) {
+  outline-width: 0.15rem;
+  box-shadow: 0 0 0 0.55rem rgb(var(--color-background)), 0 0 0 0.85rem rgba(var(--color-foreground), 0.5),
+    0 0 1.15rem 0.1rem rgba(var(--color-foreground));
+}
+
+/* Focus state for older browsers */
+@supports not selector(:has(a, b)) {
+  .visual-display-parent:focus-within .visual-display--style-swatch,
+  .visual-display-parent--thumbnail:focus-within {
+    outline-offset: 0.2rem;
+    outline: 0.15rem solid rgb(var(--color-foreground));
+    box-shadow: 0 0 0 0.55rem rgb(var(--color-background)), 0 0 0 0.85rem rgba(var(--color-foreground), 0.5),
+      0 0 1.15rem 0.1rem rgba(var(--color-foreground));
+  }
+}
+
+.visual-display-parent.disabled {
+  opacity: 0.4;
+  pointer-events: none;
+}
+
+/* Used to display the disabled dash */
+.visual-display-parent.disabled .visual-display::after {
+  display: block;
+  content: '';
+
+  /* 1.414 is not a magic number, it's the square root of 2, or the length of the diagonal */
+  width: calc(var(--visual-display__size) * 1.414);
+  border-bottom: 0.1rem solid rgba(var(--color-foreground), 0.4);
+  transform: rotate(-45deg);
+  transform-origin: left;
+}
+
+.visual-display > * {
+  display: block !important;
+  height: 100%;
+  width: 100%;
+}
+
+.visual-display__image {
+  object-fit: cover;
+}
```

Comment:
```suggestion
.visual-display__image--swatch {
  object-fit: cover;
}

.visual-display__image--thumbnail {
  object-fit: contain;
}
```

^ These classes don't exist yet.


From our sync: **Thumbnail** Images that are much taller than wide (or vice versa) will get a zoomed in look that may make it difficult to tell what the image is. Screenshot example below is showing a Fridge.  I think `contain` is better for thumbnails, even if it results in some filter images having different dimensions. However, when the images are used in a **swatch**, I agree that `cover` is the right implementation


<img width="376" alt="Screenshot 2023-10-06 at 5 39 48 PM" src="https://github.com/Shopify/dawn/assets/6284114/f871e829-efdb-4911-a5da-b1ca2dec58a4">


<img width="408" alt="Screenshot 2023-10-06 at 5 39 33 PM" src="https://github.com/Shopify/dawn/assets/6284114/f224897f-a82d-45a9-a910-8b2962a504be">
different proportions - the merchant uploaded them that way.

Reply:
I already have top-level classes signifying the style, so I can simply do:
```css
.visual-display--style-swatch .visual-display__image {
  object-fit: cover;
}

.visual-display--style-thumbnail .visual-display__image {
  object-fit: contain;
}
```